lib/std/time/Date.zig

```diff
@@ -0,0 +1,265 @@
+year: Year = 1970,
+month: Month = .January,
+day_of_month: DayOfMonth = 1, // 1 - 31
+
+const Date = @This();
+const std = @import("../std.zig");
+const s_per_day = std.time.s_per_day;
+
+const seconds_from_epoch_to_y2k = 30 * (365 * s_per_day) + 7 * s_per_day;
+
+pub const Year = i32;
+pub const DayOfMonth = u5; // 1-31
+pub const DayOfYear = u9; // 0-365
+pub const WeekOfYear = u6; // 0-53
+
+pub const Month = enum(u4) {
+    January = 1,
+    February,
+    March,
+    April,
+    May,
+    June,
+    July,
+    August,
+    September,
+    October,
+    November,
+    December,
+
+    /// Get the 3 letter abbreviation of the month
+    pub fn abbrev(month: Month) []const u8 {
+        return @tagName(month)[0..3];
+    }
+
+    /// Determine the month from the first three letters of `str`.
+    pub fn parse(str: []const u8) ?Month {
+        const str3 = if (str.len >= 3) str[0..3] else return null;
+        inline for (std.meta.fields(Month)) |field| {
+            const month = @field(Month, field.name);
+            if (std.ascii.eqlIgnoreCase(str3, field.name[0..3])) {
+                return month;
+            }
+        }
+        return null;
+    }
+
+    /// Calculate the number of days in this month.
+    pub fn daysInMonth(month: Month, is_leap: bool) DayOfMonth {
+        return if (month == .February and is_leap)
+            29
+        else
+            ([_]DayOfMonth{ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 })[@enumToInt(month) - 1];
+    }
+
+    /// Calculate the number of seconds this month is into the year.
+    pub fn secondsIntoYear(month: Month, is_leap: bool) u32 {
+        const S = s_per_day;
+        const secs_through_month = [_]u32{ 0, 31 * S, 59 * S, 90 * S, 120 * S, 151 * S, 181 * S, 212 * S, 243 * S, 273 * S, 304 * S, 334 * S };
+        const midx = @enumToInt(month) - 1;
+        return secs_through_month[midx] + @as(u32, if (is_leap and midx >= 2) S else 0);
+    }
+};
+
+pub const Weekday = enum(u3) {
+    Sunday,
```

Comment:
Weekdays should be numbered from 1 to 7 starting with monday.

Reply:
This is a POSIX vs ISO convention debate that I would rather not get into.  These algorithms are based upon what is typically done in libc.  And since Zig eventually hopes to implement it's own libc, the POSIX way made more sense to me.